### PR TITLE
Mixin magic methods

### PR DESCRIPTION
**Pull request by Artistan**
This is a good start to adding the proper mixins for magic methods within the laravel framework.
I hope this will help others understand the dynamic nature of the framework. It has helped me.

### CODE REVIEW

File: src/Illuminate/Redis/Connections/PhpRedisConnection.php [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 use Closure;
 
+/**
+ * @mixin \Predis\Client
```
Comment (it-can) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]: Is this correct? This needs to reference phpredis (\Redis) and not predis...?

Comment (Artistan) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]: /**
     * Create a new Predis connection.
     *

It is a client, the Connection class says 

    /**
     * The Predis client.
     *
     * @var \Predis\Client
     */
    protected $client;

But the PhpRedisConnection says ... both?

    /**
     * Create a new Predis connection.
     *
     * @param  \Redis  $client
     * @return void
     */
    public function __construct($client)
    {
        $this->client = $client;
    }

Comment (it-can) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]: Yeah the comment is incorrect in PhpRedisConnection

Comment (Artistan) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]: So it should be this right?

/**
 * Create a new Predis connection.
 *
 * @param  \Predis\Client  $client
 * @return void
 */

Comment (it-can) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]: I'll add a PR

Comment (it-can) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]: https://github.com/laravel/framework/pull/20235

Comment (Artistan) [inline src/Illuminate/Redis/Connections/PhpRedisConnection.php]: thanks.

**Comment by nmfzone** [comment]
Just want to notify if `@mixin` (interface) in the Manager would make IDE complaint about `method not getting implemented` when extending the Manager.

Would like to submit a PR when I have a free time.

**Pull request merged**